crates/sui-core/src/authority.rs

```diff
@@ -980,7 +973,10 @@ impl AuthorityState {
 
         // Notifies transaction manager about available input objects. This allows the transaction
         // manager to schedule ready transactions.
-        // TODO: investigate switching TransactionManager to use notify_read() on objects table.
+        //
+        // REQUIRED: this must be called after commit_certificate() (above), to ensure
```

Comment:
it may also be worth noting that we must call it before tx_guard.commit_tx() - otherwise it could be skipped due to a crash.

Reply:
Yes, will update the comment. In the hopefully next PR I will use a single `TransactionManager::commit()` call to notify available objects and "unlock" the transaction.